### PR TITLE
allow merging of string and array values for config keys

### PR DESCRIPTION
**Pull request by alcohol**
Could not merge a config key that accepted both a string or array value (default provided was a string, my value was an array). This is my proposed solution. Cannot say with confidence it is elegant or desirable to allow this however. Please close PR if you do not agree or see a valid use case for this behavior.

Also, some CS/PSR2 thingies "fixed".

### CODE REVIEW

**Comment by alcohol** [comment]
Actually, I just realized, it does not merge the string into the array. It replaces the string value with the new array value.

In short:

```
foo: bar + foo: [ baz, bez ] > foo: [ baz, bez ]
```

**Comment by alcohol** [comment]
```
foo: bar + foo: [] > foo: [ bar ]
```

Is that desired behavior?

Now it is:

```
foo: bar + foo: [] > foo: []
```

File: src/Igorw/Silex/ConfigServiceProvider.php [hunk 1]
```diff
@@ -68,7 +73,10 @@ private function mergeRecursively(array $currentValue, array $newValue)
     {
         foreach ($newValue as $name => $value) {
             if (is_array($value) && isset($currentValue[$name])) {
```
Comment (davedevelopment) [inline src/Igorw/Silex/ConfigServiceProvider.php:79]: perhaps just change this to 

if (is_array($value) && isset($currentValue[$name]) && is_array($currentValue[$name])) {

File: src/Igorw/Silex/ConfigServiceProvider.php [hunk 2]
```diff
@@ -68,7 +73,10 @@ private function mergeRecursively(array $currentValue, array $newValue)
...
-                $currentValue[$name] = $this->mergeRecursively($currentValue[$name], $value);
+                $currentValue[$name] = $this->mergeRecursively(
+                    !is_array($currentValue[$name]) ? array($currentValue[$name]) : $currentValue[$name],
```
Comment (davedevelopment) [inline src/Igorw/Silex/ConfigServiceProvider.php:81]: And change this back to how it was

**Pull request closed without merging**